crates/subspace-farmer/src/plotting.rs

```diff
@@ -94,13 +101,15 @@ impl Drop for Plotting {
 // TODO: Blocks that are coming form substrate node are fully trusted right now, which we probably
 //  don't want eventually
 /// Maintains plot in up to date state plotting new pieces as they are produced on the network.
+#[allow(clippy::too_many_arguments)]
 async fn background_plotting<T: RpcClient + Clone + Send + 'static>(
     client: T,
     plot: Plot,
     commitments: Commitments,
     object_mappings: ObjectMappings,
     farmer_metadata: FarmerMetadata,
```

Comment:
This is not that important, but I agree with clippy here. Number of arguments may be scary for an outsider. 
Should we introduce a struct, namely `FarmerData` and wrap:
- `client`
- `plot`
- `commitments`,
- `object_mappings`,
- `farmer_metadata`
with it?

Reply:
Yes, we should, but I was lazy

Reply:
I can try today, I have time :)

Reply:
Did it, should I push my changes into here, or create another branch?
Also, I did not include `client` in the struct, seemed like it does not belong to `FarmerData`. But I can include it if you like?